core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1108,6 +1108,46 @@ public Builder setDefaultPartitionSpec(int specId) {
       return this;
     }
 
+    Builder removeUnusedSpecs(Iterable<PartitionSpec> specsToRemove) {
```

Comment:
Sorry not completely following, do we need this? Why couldn't we always just use `removeUnusedSpecsById`?